### ISSUE TITLE
Expose domain of functions

### ISSUE THREAD

**Issue by dpsanders** [JuliaIntervals/IntervalArithmetic.jl#661]
Currently the user has no way of knowing which domain a function is defined over. 

E.g.
```
domain(sqrt) = 0..Inf
```
or `domain_interval(sqrt)`.

**Comment by OlivierHnt** [comment]
I do not think we should do this. Sometimes the domain is not closed, eg for `log`.

However I think this should appear in the docstring of `sqrt`.

**Comment by dpsanders** [comment]
Good point about the domain for `log`. 
Nonetheless I think there ideally should be *some* way for the user to obtain this information programatically.

**Comment by OlivierHnt** [comment]
In the end, this is truly a math question and not interval related.
How the domain relates to intervals (which is what matters for the library) depends on the function (what we do for `sqrt` is different than what we do for other functions).

That being said, we can leave this issue open for a little while and see if someone comes up with something.
But I suspect that this is a matter of documentation.

**Comment by dpsanders** [comment]
The reason I was thinking about this is that I wanted to do something like `Piecewise(sqrt, extend=0)` meaning "extend sqrt with 0 outside its domain of definition", but that requires programmatic access to the domain.

**Comment by OlivierHnt** [comment]
Oh I see, interesting!

Then `Piecewise(log, extend = 0)` would also work I think, or even `Piecewise(asec; extend = 0)`.
The logic for this feature is more like a `isindomain` function

Something in the vein of
```
isindomain(::typeof(sqrt), x::Float64) = 0 <= x

isindomain(::typeof(sqrt), x::Interval) = issubset_interval(x, interval(0, Inf))
```

**Comment by OlivierHnt** [comment]
Btw since PR #689 was merged last year, we have a `Domain` type which can handle closed / open intervals.
So I believe we have enough to implement your suggestion:

```Julia
Piecewise(sqrt; extension = x -> 0) # extend `sqrt` by 0 beyond its domain of definition.
```
